doc/man7/EVP_KDF-PBKDF2.pod

```diff
@@ -53,6 +53,12 @@ The checks performed are:
 
 =item - the derived key length is at least 112 bits.
 
+=item - the password length is at least minimum length.
+
+Note minimum password length is a configurable compile time constant,
+defaulting to zero for the default provider and 8 for the FIPS
```

Comment:
Why is it not 14 bytes?

Reply:
14 seems like the minimal value to me, since there are not normally even 112 bits due to alphanumeric characters being used.

Reply:
Because passwords are meant to be usable by humans. Although various SPs do recommend to enable unrestricted autofill and password managers. FIPS 140-3 IG D.N SP 800-132 Password-Based Key Derivation for Storage Applications (https://csrc.nist.gov/csrc/media/Projects/cryptographic-module-validation-program/documents/fips%20140-3/FIPS%20140-3%20IG.pdf)

> The strength of the Data Protection Key is based on the strength of the Password and/or Passphrase used in key derivation. SP 800-132 does not impose any strictly defined requirements on the strength of a password. It says that “passwords should be strong enough so that it is infeasible for attackers to get access by guessing a password.” Therefore, the vendor shall document in the module’s Security Policy the length of a password/passphrase used in key derivation and establish an upper bound for the probability of having this parameter guessed at random. This probability shall take into account not only the length of the password/passphrase, but also the difficulty of guessing it. The decision on the minimum length of a password used for key derivation is the vendor’s, but the vendor shall at a minimum informally justify the decision. 

There is no specified number. Value 8 has been justified before. We can pick a different number. Many modules choose 8, some choose higher. Password entropy depends on using cased letter and special characters, but security analysis of pins and password generally that human made up password have low entropy.

I like 12 and 16 as well, because of:
```
$ openssl rand -base64 9
GnnYNWjdIQwN
$ openssl rand -base64 12
qkt3LDGQvgDiyflV
```

Reply:
I would be against making the default higher than the BC value of 14.  That would break things for quite a few people that interop with BC.

Reply:
OpenSSLs from major distros:
- [CMVP 4794](https://csrc.nist.gov/projects/cryptographic-module-validation-program/certificate/4794) "The minimum length of the password or passphrase accepted by the module is 8 characters."
- [CMVP 4779](https://csrc.nist.gov/projects/cryptographic-module-validation-program/certificate/4779) "The length of the password or passphrase shall be at least 8 characters, and shall consist of lowercase, uppercase, and numeric characters."
- [CMVP 4746](https://csrc.nist.gov/projects/cryptographic-module-validation-program/certificate/4746) "The minimum length of the password or passphrase accepted by the module is 8 characters."
- [CMVP 4725](https://csrc.nist.gov/CSRC/media/projects/cryptographic-module-validation-program/documents/security-policies/140sp4725.pdf) "The length of the password or passphrase shall be of at least 20 characters (this is verified by the module to determine the service is approved), and shall consist of lower-case, uppercase and numeric characters."

BouncyCastle FIPS JAVA API:
- [CMVP 4743](https://csrc.nist.gov/projects/cryptographic-module-validation-program/certificate/4743) "In approved mode the module enforces that any password used must encode to at least 14 bytes
(112 bits)"

So setting higher than 8 will create interop problems with distro provided FIPS modules, setting to lower than 14 will have interop problems with BC-FJA, and Suse likes 20 character long passwords.

Reply:
So sticking with 8 as default seems reasonable..

Reply:
I assume this would be the value the indicator compares against..

Reply:
I believe all of this is already part of the `"pkcs5" (OSSL_KDF_PARAM_PKCS5) <integer>` param and thus `"fips-indicator" (OSSL_KDF_PARAM_FIPS_APPROVED_INDICATOR) <integer>` and the related pbkdf2-lower-bound-check fips module runtime config option.